src/matlab2tikz.m

```diff
@@ -407,9 +407,19 @@ function matlab2tikz(varargin)
 end
 % ==============================================================================
 function fid = fileOpenForWrite(m2t, filename)
-    encoding = switchMatOct({'native', m2t.cmdOpts.Results.encoding}, {});
```

Comment:
What's the reason for changing the `switchMatOct` to the more verbose `if-else`?
Personally, I prefer such things in a 
```matlab
switch getEnvironment
    case 'MATLAB'
        ...
    case 'Octave'
        ...
    otherwise
        errorUnknownEnvironment()
end
```
since that is how it's called in most parts of the code and I generally dislike `strcmp`/`strcmpi` for stylistic reasons.